File: 38-Count-and-Say/38-Count-and-Say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        def say(n):
            count = 1
            ans = ""
            curr = n[0]
            for i in n[1:]:
                if i  == curr:
                    count += 1
                else:
                    ans = ans + str(count) + str(curr)
                    curr = i
                    count = 1
            
            ans = ans + str(count) + str(curr)
            return ans
        ans = "1"
        for i in range(n-1):
            ans = say(ans)
        return ans
        
    

File: 38-Count-and-Say/38-Count-and-Say.java
class Solution {
    public String countAndSay(int n) {
        String ans = "1";
        for(int i = 0; i < n -1; i++){
            ans = say(ans);
        }
        return ans;
    }
    private String say(String s){
        int count = 1;
        char curr = s.charAt(0);
        String ans = "";
        for(int i = 1; i < s.length(); i++){
            if(curr == s.charAt(i)){
                count += 1;
            } else{
                ans = ans + String.valueOf(count);
                ans = ans + String.valueOf(curr);
                curr = s.charAt(i);
                count = 1;
            }
        }
        ans = ans + String.valueOf(count);
        ans = ans + String.valueOf(curr);
        return ans;

    }
}